libs/blocks/merch/merch.js

```diff
@@ -357,13 +357,20 @@ async function openExternalModal(url, getModal) {
 
 const isInternalModal = (url) => /\/fragments\//.test(url);
 
-export async function openModal(e, url, offerType) {
+export async function openModal(e, url, offerType, hash) {
   e.preventDefault();
   e.stopImmediatePropagation();
   const { getModal } = await import('../modal/modal.js');
   await import('../modal/modal.merch.js');
   const offerTypeClass = offerType === OFFER_TYPE_TRIAL ? 'twp' : 'crm';
   let modal;
+  if (hash) {
```

Comment:
I think we should require gwp to author hashes in the checkout-link.xlsx, generated hashes might not be good w.r.t SEO, when linked from other pages.

If the hash is present, we can use it to handle the modal, if not, we can keep the current behaviour.

GWP can progressively add it and publish the file.